nixos/modules/services/networking/syncthing.nix

```diff
@@ -10,6 +10,28 @@ let
   settingsFormat = pkgs.formats.json { };
   cleanedConfig = converge (filterAttrsRecursive (_: v: v != null && v != {})) cfg.settings;
 
+  # syncthing supports serving the gui over unix sockets. if that happens, the
+  # api is serverd over the unix socket as well. this module uses the api to
+  # merge the configuration. curl is used for this and supports unix sockets.
+  # however the arguments are slightly different from an http request. the
+  # function `curlAddressArgs` returns a given path the correct string to make
+  # curl connect over either http(s) or unix. string which tells curl how to
+  # connect to the address. this works for https and unix.
+  curlAddressArgs = path: let
+    matchResult = builtins.match "^unix://(.*)" cfg.guiAddress;
```

Comment:
I don't know about requiring a non-standard `unix://` prefix. The user could also easily write `unix://tmp/st.sock` which would match on `tmp/st.sock` (not an absolute path and will lead to subtle confusing errors). Couldn't we just check if the address starts with a `/`? We would also be making sure the address we get is an absolute path, and that's very important here.

Reply:
if we really want to do validity checks on the url we should do that with assertions. syncthing's --gui-address override needs the unix://prefix, otherwise it will be ignored. however, given how the syncthing docs read (`UNIX socket location (/var/run/st.sock)`) and the arguments you brought up, I'm in favor of inserting the unix:// when assembling the cli arguments and having the user specify an absolute path.